### PR TITLE
[pybind] add pybind for kaldi::GeneralMatrix (#93)

### PR DESCRIPTION
**Pull request by yaguanghu**
create pybind binding for kaldi::GeneralMatrix, which is used in nnet3 computing.

### CODE REVIEW

**Comment by csukuangfj** [comment]
@danpovey @naxingyu  @songmeixu 
please have a review

File: src/pybind/matrix/sparse_matrix_pybind.cc [hunk 1]
```diff
@@ -27,6 +27,19 @@ using namespace kaldi;
 // so we need to wrap GeneralMatrix
 
 void pybind_sparse_matrix(py::module& m) {
+  {
+    using PyClass = SparseMatrix<BaseFloat>;
+    py::class_<PyClass>(
+        m, "SparseMatrix",
+        "This class is defined for sparse matrix type.")
+        .def(py::init<>())
+        .def(py::init<const MatrixBase<BaseFloat> &>())
```
Comment (naxingyu) [inline src/pybind/matrix/sparse_matrix_pybind.cc]: py::arg("other")

File: src/pybind/matrix/sparse_matrix_pybind.cc [hunk 2]
```diff
@@ -27,6 +27,19 @@ using namespace kaldi;
...
+        .def(py::init<int32, int32>())
```
Comment (naxingyu) [inline src/pybind/matrix/sparse_matrix_pybind.cc]: py::arg("row"), py::arg("col")

File: src/pybind/matrix/sparse_matrix_pybind.cc [hunk 3]
```diff
@@ -37,7 +50,14 @@ void pybind_sparse_matrix(py::module& m) {
         "and write a single object type.  It is useful for neural-net training "
         "targets which might be sparse or not, and might be compressed or not.")
         .def(py::init<>())
-        // TODO(fangjun): wrap other methods when needed
+        .def(py::init<const MatrixBase<BaseFloat> &>())
```
Comment (naxingyu) [inline src/pybind/matrix/sparse_matrix_pybind.cc]: py::arg("other")

File: src/pybind/matrix/sparse_matrix_pybind.cc [hunk 4]
```diff
@@ -37,7 +50,14 @@ void pybind_sparse_matrix(py::module& m) {
...
+        .def(py::init<const SparseMatrix<BaseFloat> &>())
```
Comment (naxingyu) [inline src/pybind/matrix/sparse_matrix_pybind.cc]: py::arg("other")

**Comment by danpovey** [comment]
Thanks.. when the PR is OK with Fangjun, it's OK with to merge; and Meixu
can do it as he has access (I don't want to slow things down).

**Comment by csukuangfj** [comment]
+2

**Comment by csukuangfj** [comment]
@songmeixu

**Comment by danpovey** [comment]
its OK I'll merge this time..

**Pull request merged**